provisioning/service/src/Authentication/SecurityHelper.cs

```diff
@@ -2,25 +2,16 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Diagnostics;
 using System.Globalization;
 
 namespace Microsoft.Azure.Devices
 {
     internal static class SecurityHelper
     {
-        public static void ValidateServiceHostName(string serviceHostName, string serviceName)
+        internal static void ValidateServiceHostName(string serviceHostName, string shareAccessSignatureName)
         {
-            if (string.IsNullOrWhiteSpace(serviceHostName))
-            {
-                throw new ArgumentNullException(nameof(serviceHostName));
-            }
-
-            if (string.IsNullOrWhiteSpace(serviceName))
-            {
-                throw new ArgumentNullException(nameof(serviceName));
-            }
```

Comment:
These checks are performed in ServiceConnectionStringBuilder where connection string is parsed and broken in different parameters.